src/ucs/datastruct/lru.h

```diff
@@ -0,0 +1,50 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2001-2023. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef __LRU_H__
+#define __LRU_H__
+
+#include <ucs/type/status.h>
+#include <stddef.h>
+
+typedef struct ucs_lru *ucs_lru_h;
+
+/**
+ * @brief Init a new LRU cache instance.
+ *
+ * @param [in] capacity  Max number of elements to store.
+ *
+ * @return Pointer to the allocated cache. In case of error, NULL returned.
+ */
+ucs_lru_h ucs_lru_init(size_t capacity);
+
+/**
+ * @brief Destroy LRU cache.
+ *
+ * @param [in] lru  Handle to the LRU cache.
+ */
+void ucs_lru_destroy(ucs_lru_h lru);
+
+/**
+ * @brief Insert or update an element in the cache.
+ *
+ * @param [in] lru  Handle to the LRU cache.
+ * @param [in] key  Element's key.
+ *
+ */
+void ucs_lru_put(ucs_lru_h lru, void *key);
+
+/**
+ * @brief Get list of elements from the cache.
+ *
+ * @param [in]  lru   Handle to the LRU cache.
+ * @param [out] array Pointer to an allocated array to store elements.
+ *                    Assume it contains at least 'lru->capacity' elements.

```

Comment:
1. `Assume` -> must
2. Better describe in terms of bytes (how much memory is required) and not in terms of "elements capacity".